gen/cpp/sym/factors/imu_manifold_preintegration_update.h

```diff
@@ -0,0 +1,845 @@
+// -----------------------------------------------------------------------------
+// This file was autogenerated by symforce from template:
+//     function/FUNCTION.h.jinja
+// Do NOT modify by hand.
+// -----------------------------------------------------------------------------
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/rot3.h>
+
+namespace sym {
+
+/**
+ * An internal helper function to update a set of preintegrated IMU measurements with a new pair of
+ * gyroscope and accelerometer measurements. Returns the updated preintegrated measurements.
+ *
+ * When integrating the first measurement, DR should be the identity, Dv, Dp, covariance, and the
+ * derivatives w.r.t. to bias should all be 0.
+ *
+ * Args:
+ *     DR (sf.Rot3): Preintegrated change in orientation
+ *     Dv (sf.V3): Preintegrated change in velocity
+ *     Dp (sf.V3): Preintegrated change in position
+ *     covariance (sf.M99): Covariance of [DR's tangent space, Dv, Dp]
+ *     DR_D_gyro_bias (sf.M33): Derivative of DR w.r.t. gyro_bias
+ *     Dv_D_gyro_bias (sf.M33): Derivative of Dv w.r.t. gyro_bias
+ *     Dv_D_accel_bias (sf.M33): Derivative of Dv w.r.t. accel_bias
+ *     Dp_D_gyro_bias (sf.M33): Derivative of Dp w.r.t. gyro_bias
+ *     Dp_D_accel_bias (sf.M33): Derivative of Dp w.r.t. accel_bias
+ *     gyro_bias (sf.V3): Initial guess for the gyroscope measurement bias
+ *     accel_bias (sf.V3): Initial guess for the accelerometer measurement bias
+ *     gyro_cov (sf.M33): The covariance of the gyroscope measurement
+ *     accel_cov (sf.M33): The covariance of the accelerometer measurement
```

Comment:
It saves a few.
```
-  // Total ops: 1884
+  // Total ops: 1700
```
Though, this is a function that might get called 100 times or whatever per factor (depending on how many factors you have).

I suppose it's simply not too many because they're only 3x3 matrices.

Still, I don't think having multiple versions would be too annoying because they can just be additional methods on the class.

Reply:
It would be good to have this as a TODO somewhere in the code

Reply:
Noting that we've now made these diagonal, per the other thread here: https://github.com/symforce-org/symforce/pull/279#discussion_r1081978527